### PR TITLE
:bug: Fix getUserBalances by using standard fetch instead of axios

### PR DESCRIPTION
**Pull request by ghivert**
### Description of change

An outdated version of `axios` was causing an `unexpected end of file` bug. Switching to `fetch` is enough to make the bug go away.
It's using `node-fetch` as a polyfill waiting for Node 18 to be broadly available. When it will be, we'll be able to remove `node-fetch` and still using the exact same codebase.

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [x] `yarn lint` passes with this change
- [x] `yarn test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

**Review by Rubilmax — COMMENTED** [review]
We could require node `>=18` in `package.json`, considering that nodejs v18 is widely available?

**Comment by ghivert** [comment]
We talked with Julien, and it's not as straightforward as I thought, he'll check on the WE

File: src/utils/getUserRewards.ts [hunk 1]
```diff
@@ -160,21 +160,23 @@ const computeIndex = async (
...
       variables: { user, block },
-    })
-    .then((r) => {
-      if (!r.data?.data) throw Error(JSON.stringify(r.data.errors));
-      if (!r.data.data.user) return undefined;
-      return formatGraphBalances(r.data.data.user);
-    });
+    }),
+  });
+  if (!res.ok) throw Error(res.status.toString());
```
Comment (julien-devatom) [inline src/utils/getUserRewards.ts:172]: What if there is a non 200 resp but with more information in the body? here for a subgraph error, we have more information in `res.data.errors` for example

**Review by julien-devatom — CHANGES_REQUESTED** [review]

**Review by julien-devatom — APPROVED** [review]
LGTM

**Pull request merged**